src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+
+      Log::Info << "Training model with L-BFGS optimizer." << endl;
+
+      // This will train the model.
+      model->Train(trainingSet, labels, numClasses, lbfgsOpt);
+    }
+    // This optimizer is only compiled if OpenMP is used.
+    #ifdef HAS_OPENMP
+    else if (optimizerType == "psgd")
+    {
+      const double step_size = CLI::GetParam<double>("step_size");
+      const bool shuffle = !CLI::HasParam("shuffle");
+
+      ens::ConstantStep decayPolicy(step_size);
+
+      ens::ParallelSGD<ens::ConstantStep> psgdOpt(maxIterations, std::ceil(
+        (float)trainingSet.n_cols / omp_get_max_threads()), tolerance, shuffle,
+        decayPolicy);
+
+      Log::Info << "Training model with ParallelSGD optimizer." << endl;
+
+      // This will train the model.
+      model->Train(trainingSet, labels, numClasses, psgdOpt);
+    }
+    #endif
+
+    // If OpenMP is not specified but optimizer is psgd.
+    #ifndef HAS_OPENMP
+    else if (optimizerType == "psgd")
+    {
+      Log::Fatal << "OpenMP is not specified cannot use ParallelSGD" << endl;
+    }
+
+    RequireParamInSet<string>("optimizer", { "lbfgs" },
+        true, "unknown optimizer");
+    #endif
+  }
+  if (CLI::HasParam("test"))
+  {
+    // Get the test dataset, and get predictions.
+    testSet = std::move(CLI::GetParam<arma::mat>("test"));
+    arma::Row<size_t> predictions;
+    size_t trainingDimensionality;
+
+    // Set the dimensionality according to fitintercept.
+    if (intercept && optimizerType == "lbfgs")
+      trainingDimensionality = model->Parameters().n_rows - 1;
+    else
+      trainingDimensionality = model->Parameters().n_rows;
+
+    // Checking the dimensionality of the test data.
+    if (testSet.n_rows != trainingDimensionality)
+    {
+      Log::Fatal << "Test data dimensionality (" << testSet.n_rows << ") must "
+          << "be the same as the dimensionality of the training data ("
+          << trainingDimensionality << ")!" << endl;
+    }
+
+    // Save class score, if desired.
+    if (CLI::HasParam("score"))
+    {
+      Log::Info << "Calculating class score of points in '"
+          << CLI::GetPrintableParam<arma::mat>("test") << "'." << endl;
+      arma::mat score;
+      model->Classify(testSet, score);
+      CLI::GetParam<arma::mat>("score") = std::move(score);
+    }
+
+    model->Classify(testSet, predictions);
+
+    // Calculate accuracy, if desired.
+    if (CLI::HasParam("test_labels"))
+    {
+      arma::Row<size_t> testLabels =
+        std::move(CLI::GetParam<arma::Row<size_t>>("test_labels"));
+
+      if (testSet.n_cols != testLabels.n_elem)
+      {
+        Log::Fatal << "Test data given with " << PRINT_PARAM_STRING("test")
+            << " has " << testSet.n_cols << " points, but labels in "
+            << PRINT_PARAM_STRING("test_labels") << " have "
+            << testLabels.n_elem << " labels!" << endl;
+      }
+
+      vector<size_t> bingoLabels(numClasses, 0);
```

Comment:
Maybe it's clearer to call this `correctClassCounts` instead of `bingoLabels`?  Up to you.